### PR TITLE
Remove aliases of various api.types APIs from utils.dtypes.

### PR DESCRIPTION
**Pull request by vyasr**
Continuation of #8308 that moves all imports of standard dtype utilities to use `cudf.api.types` or `cudf.core.dtypes` rather than `cudf.util.dtypes`.

### CODE REVIEW

**Comment by vyasr** [comment]
rerun tests

**Comment by vyasr** [comment]
rerun tests

**Comment by vyasr** [comment]
rerun tests

**Comment by vyasr** [comment]
rerun tests

**Comment by vyasr** [comment]
rerun tests

File: python/cudf/cudf/utils/dtypes.py [hunk 1]
```diff
@@ -11,27 +11,6 @@
...
-    is_interval_dtype,
-    is_list_dtype,
-    is_list_like,
-    is_numeric_dtype as is_numerical_dtype,
-    is_scalar,
-    is_string_dtype,
-    is_struct_dtype,
-    is_timedelta_dtype,
-    pandas_dtype,
-)
```
Comment (galipremsagar) [inline python/cudf/cudf/utils/dtypes.py:34]: ❤️

Comment (galipremsagar) [reply python/cudf/cudf/utils/dtypes.py:34]: Helps us to get rid of circular import issues to an extent with methods related to this file

Comment (vyasr) [reply python/cudf/cudf/utils/dtypes.py:34]: Yes absolutely.

**Comment by vyasr** [comment]
rerun tests

**Comment by vyasr** [comment]
@gpucibot merge

**Comment by vyasr** [comment]
rerun tests

**Review by shwina — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**